tests/sentry/api/endpoints/test_auth_index.py

```diff
@@ -420,29 +417,14 @@ def test_superuser_no_sso_user_no_password_or_u2f_self_hosted(self):
             assert response.status_code == 400
             assert response.data["detail"]["code"] == MISSING_PASSWORD_OR_U2F_CODE
 
+    @override_settings(SENTRY_SELF_HOSTED=False)
     # su form is disabled by overriding local dev setting which skip checks
     @mock.patch("sentry.api.endpoints.auth_index.DISABLE_SSO_CHECK_FOR_LOCAL_DEV", True)
-    @with_feature("organizations:u2f-superuser-form")
-    def test_superuser_no_sso_user_has_password_su_form_off_saas(self):
```

Comment:
I combined these two tests into one b/c:
`test_superuser_no_sso_user_has_password_su_form_off_saas`
seems like a subset of
`test_superuser_no_sso_user_no_password_or_u2f_sso_disabled_local_dev_saas`,
where if the latter passes, the former should obviously as well